client-lib/go/client/jvs_client.go

```diff
@@ -50,12 +59,50 @@ func NewJVSClient(ctx context.Context, config *JVSConfig) (*JVSClient, error) {
 	}
 
 	return &JVSClient{
-		config: config,
-		keys:   cached,
+		config:           config,
+		keys:             cached,
+		forbidBreakglass: config.ForbidBreakglass,
 	}, nil
 }
 
 // ValidateJWT takes a jwt string, converts it to a JWT, and validates the signature.
 func (j *JVSClient) ValidateJWT(jwtStr string) (*jwt.Token, error) {
+	// Handle Break-glass tokens.
+	if strings.HasSuffix(jwtStr, UnsignedPostfix) {
+		token, err := jwt.Parse([]byte(jwtStr), jwt.WithVerify(false))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse jwt %s: %w", jwtStr, err)
+		}
+		valid, err := j.unsignedTokenValidAndAllowed(token)
+		if !valid {
+			return nil, fmt.Errorf("token unsigned and could not be validated: %w", err)
+		}
+		return &token, nil
+	}
 	return jvscrypto.ValidateJWT(j.keys, jwtStr)
 }
+
+func (j *JVSClient) unsignedTokenValidAndAllowed(token jwt.Token) (bool, error) {
+	if j.forbidBreakglass {
+		return false, fmt.Errorf("break glass tokens is forbidden, denying")
+	}
+	justs, ok := token.Get("justs")
+	if !ok {
+		return false, fmt.Errorf("can't find 'justs' in claims, denying")
+	}
+	justsBytes, err := json.Marshal(justs)
+	if err != nil {
+		return false, fmt.Errorf("failed to marshal 'justs', denying")
+	}
+	var justifications []*jvspb.Justification
+	err = json.Unmarshal(justsBytes, &justifications)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal 'justs', denying")
+	}
+	for _, justification := range justifications {
+		if justification.GetCategory() == BreakglassCategory {
+			return true, nil
```

Comment:
Should we expect all justifications to have `breakglass` category?

Reply:
The logic in java client only check one justification to have a `breakglass` category.
https://github.com/abcxyz/jvs/blob/3b5e51097c09a19a4f2ecfff5be9c6ccb1fcdc3e/client-lib/java/src/main/java/com/abcxyz/jvs/JvsClient.java#L78

Reply:
@raserva I wonder what's the meaning if not all justification has `breakglass` category?

Reply:
The expectation is to ensure that this is intended to be a breakglass token, and not just an unsigned token

Reply:
if we didn't have some sort of sentinel value, then any unsigned JWT would be treated as a breakglass token, even if it was not intended to be.

Reply:
My question is whether we should ensure **all** categories are `breakglass` or just **one-of** them.

Reply:
This logic looks over all justifications and if any are breakglass, it return true. Only if it never finds one does it return false

Reply:
here it returns nil if it finds one, but doesn't force all categories to be breakglass

Reply:
Ack. We need to document this behavior in the user manual.